src/storage-client/src/types/connections.rs

```diff
@@ -349,13 +376,59 @@ impl KafkaConnection {
         }
 
         let mut context = BrokerRewritingClientContext::new(context);
+        // TODO(guswynn): factor this out
         for broker in &self.brokers {
-            if let Some(aws_privatelink) = &broker.aws_privatelink {
-                context.add_broker_rewrite(
-                    &broker.address,
-                    &mz_cloud_resources::vpc_endpoint_name(aws_privatelink.connection_id),
-                    aws_privatelink.port,
-                );
+            match &broker.tunnel {
+                Tunnel::Direct => {
+                    // By default, don't override broker address lookup.
+                }
+                Tunnel::AwsPrivateLink(aws_privatelink) => {
+                    context.add_broker_rewrite(
+                        &broker.address,
+                        &mz_cloud_resources::vpc_endpoint_name(aws_privatelink.connection_id),
+                        aws_privatelink.port,
+                    );
+                }
+                Tunnel::Ssh {
+                    connection_id,
+                    connection,
+                } => {
+                    // Setup the config...
+                    let secret = connection_context
+                        .secrets_reader
+                        .read(*connection_id)
+                        .await?;
+                    let key_set = SshKeyPairSet::from_bytes(&secret)?;
+                    let key_pair = key_set.primary().clone();
+                    let ssh_tunnel_config = SshTunnelConfig {
+                        host: connection.host.clone(),
+                        port: connection.port,
+                        user: connection.user.clone(),
+                        key_pair,
+                    };
+
+                    // ...extract the broker address pieces...
+                    use std::net::ToSocketAddrs;
+                    let broker_addr = broker
+                        .address
+                        .to_socket_addrs()?
+                        .next()
+                        .context("BROKER is not address:port")?;
+                    let broker_host = broker_addr.ip().to_string();
```

Comment:
Wait, this doesn't look right! I think you just want to split on `:`. We don't want to be doing DNS resolution here, right, because the host will be private? I'm 99% sure we want to be passing the hostname through to `connect` so that SSH can resolve the hostname remotely, on the SSH bastion host.

Reply:
@benesch thought I could get away with this, but you are right! didnt realize that we tosocketaddr's resolves the ip :(